src/main/java/bio/terra/service/snapshot/flight/delete/DeleteSnapshotPolicyStep.java

```diff
@@ -0,0 +1,45 @@
+package bio.terra.service.snapshot.flight.delete;
+
+import bio.terra.common.exception.FeatureNotImplementedException;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.flight.DatasetWorkingMapKeys;
+import bio.terra.service.job.DefaultUndoStep;
+import bio.terra.service.policy.PolicyService;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DeleteSnapshotPolicyStep extends DefaultUndoStep {
```

Comment:
Are we sure that this step should have the default undo behavior (that is, return a success on undo)?  I would have thought that undoing the deletion of a snapshot's policy would be to restore that policy.

Reply:
Oh now that I think of it, that behavior makes sense to me. We wouldn't want the case where a snapshot still exists (because the delete failed) but its policy is gone.